pkg/issuer/vault/setup.go

```diff
@@ -0,0 +1,54 @@
+package vault
+
+import (
+	"context"
+
+	"github.com/golang/glog"
+	"github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha1"
+	"k8s.io/api/core/v1"
+)
+
+const (
+	successVaultVerified = "VaultVerified"
+	messageVaultVerified = "Vault verified"
+
+	messageVaultClientInitFailed         = "Failed to initialize Vault client: "
+	errorVaultClientInitFailed           = "ErrVaultClientInit"
+	messageVaultHealthCheckFailed        = "Failed to call Vault health check: "
+	errorVaultHealthCheckFailed          = "ErrVaultHealthCheck"
+	messageVaultStatusVerificationFailed = "Vault is not initialized or is sealed: "
+	errorVaultStatusVerificationFailed   = "ErrVaultStatus"
```

Comment:
You can probably replace all the `ErrValue*` values with a simple `VaultError`. The message can convey the extra detail.